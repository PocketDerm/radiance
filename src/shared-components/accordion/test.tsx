import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Accordion from './index';

const testAccordion = {
  title: <div>title</div>,
  isOpen: false,
  onClick: (): void => undefined,
  children: <div>expansion</div>,
};

/* eslint-disable react/jsx-props-no-spreading */
describe('<Accordion />', () => {
  test('renders regular accordion', () => {
    const component = renderer.create(<Accordion {...testAccordion} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders no border accordion', () => {
    const component = renderer.create(
      <Accordion noBorder {...testAccordion} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders disabled accordion', () => {
    const component = renderer.create(
      <Accordion disabled {...testAccordion} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('invokes onClick when title is clicked', () => {
    const spy = jest.fn();
    const component = shallow(<Accordion {...testAccordion} onClick={spy} />);

    component.find('[role="button"]').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
/* eslint-enable react/jsx-props-no-spreading */
