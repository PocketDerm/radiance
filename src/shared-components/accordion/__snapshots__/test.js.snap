// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders disabled accordion 1`] = `
.emotion-0 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-6 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

.emotion-8 {
  border: 1px solid #ededf0;
  width: 100%;
  opacity: 0.4;
  background-color: #ededf0;
  border-color: #c3c0cd;
}

.emotion-8:not(:last-child) {
  border-bottom: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: not-allowed;
}

<div
  className="emotion-8 emotion-9"
  disabled={true}
>
  <div
    className="emotion-4 emotion-5"
    disabled={true}
    onClick={[Function]}
  >
    <div
      className="emotion-0 emotion-1"
    >
      <div>
        title
      </div>
    </div>
    <div
      className="emotion-2 emotion-3"
    >
      <<svg viewBox="0 0 10 10" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Chevron</title>
  <path d="M3.00025,10.00025 C2.74425,10.00025 2.48825,9.90225 2.29325,9.70725 C1.90225,9.31625 1.90225,8.68425 2.29325,8.29325 L5.58625,5.00025 L2.29325,1.70725 C1.90225,1.31625 1.90225,0.68425 2.29325,0.29325 C2.68425,-0.09775 3.31625,-0.09775 3.70725,0.29325 L7.70725,4.29325 C8.09825,4.68425 8.09825,5.31625 7.70725,5.70725 L3.70725,9.70725 C3.51225,9.90225 3.25625,10.00025 3.00025,10.00025"/>
</svg>

        fill="#332e54"
        height={8}
        rotate={0}
        width={8}
      />
    </div>
  </div>
  <div
    className="emotion-6 emotion-7"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders no border accordion 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.emotion-0 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-6 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

.emotion-8 {
  width: 100%;
}

.emotion-8:not(:last-child) {
  border-bottom: none;
}

<div
  className="emotion-8 emotion-9"
  disabled={false}
>
  <div
    className="emotion-4 emotion-5"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="emotion-0 emotion-1"
    >
      <div>
        title
      </div>
    </div>
    <div
      className="emotion-2 emotion-3"
    >
      <<svg viewBox="0 0 10 10" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Chevron</title>
  <path d="M3.00025,10.00025 C2.74425,10.00025 2.48825,9.90225 2.29325,9.70725 C1.90225,9.31625 1.90225,8.68425 2.29325,8.29325 L5.58625,5.00025 L2.29325,1.70725 C1.90225,1.31625 1.90225,0.68425 2.29325,0.29325 C2.68425,-0.09775 3.31625,-0.09775 3.70725,0.29325 L7.70725,4.29325 C8.09825,4.68425 8.09825,5.31625 7.70725,5.70725 L3.70725,9.70725 C3.51225,9.90225 3.25625,10.00025 3.00025,10.00025"/>
</svg>

        fill="#332e54"
        height={8}
        rotate={0}
        width={8}
      />
    </div>
  </div>
  <div
    className="emotion-6 emotion-7"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders regular accordion 1`] = `
.emotion-8 {
  border: 1px solid #ededf0;
  width: 100%;
}

.emotion-8:not(:last-child) {
  border-bottom: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.emotion-0 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.emotion-6 {
  max-height: 0px;
  overflow: hidden;
  -webkit-transition: max-height 350ms ease-in-out;
  transition: max-height 350ms ease-in-out;
}

<div
  className="emotion-8 emotion-9"
  disabled={false}
>
  <div
    className="emotion-4 emotion-5"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="emotion-0 emotion-1"
    >
      <div>
        title
      </div>
    </div>
    <div
      className="emotion-2 emotion-3"
    >
      <<svg viewBox="0 0 10 10" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <title>Chevron</title>
  <path d="M3.00025,10.00025 C2.74425,10.00025 2.48825,9.90225 2.29325,9.70725 C1.90225,9.31625 1.90225,8.68425 2.29325,8.29325 L5.58625,5.00025 L2.29325,1.70725 C1.90225,1.31625 1.90225,0.68425 2.29325,0.29325 C2.68425,-0.09775 3.31625,-0.09775 3.70725,0.29325 L7.70725,4.29325 C8.09825,4.68425 8.09825,5.31625 7.70725,5.70725 L3.70725,9.70725 C3.51225,9.90225 3.25625,10.00025 3.00025,10.00025"/>
</svg>

        fill="#332e54"
        height={8}
        rotate={0}
        width={8}
      />
    </div>
  </div>
  <div
    className="emotion-6 emotion-7"
  >
    <div>
      <div>
        expansion
      </div>
    </div>
  </div>
</div>
`;
